libyul/YulString.h

```diff
@@ -81,8 +80,8 @@ class YulStringRepository: boost::noncopyable
 	}
 	static constexpr std::uint64_t emptyHash() { return 14695981039346656037u; }
 private:
-	std::vector<std::shared_ptr<std::string>> m_strings;
-	std::unordered_multimap<std::uint64_t, size_t> m_hashToID;
+	std::vector<std::shared_ptr<std::string>> m_strings {std::make_shared<std::string>()};
+	std::unordered_multimap<std::uint64_t, size_t> m_hashToID {{emptyHash(), 0}};
```

Comment:
Actually it's really hard to follow what happens here. Perhaps we should even make this more verbose and create a constructor
```
YulStringRepository() { insert(string()); }
```
and create an `insert` function that handles most of `stringToHandle` except for the empty check?

Reply:
Do you really think that initializing a map is *that* hard to read? Especially if we move it to a member-initializer here I think it's quite readable actually... Or do you mean why this and ``m_strings`` is initialized that way? If that's the case I think a comment might help more than creating another function...

Reply:
As soon as I see `{{`, I start wondering about how deeply nested everything is. We should perhaps not discuss this on this PR, but I think adding some code that clarifies that this initialization satisfy the invariant about the empty string would help readability.